### PR TITLE
6597 : Lower case coordinate systems in WMS capabilities are not supported

### PR DESCRIPTION
**Pull request by MuhweziDeo**
## Description

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix

**What is the current behavior?**
A "ccordinate system not supported error" is shown

#6597 

**What is the new behavior?**
The layer is added correctly.

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Review by allyoucanmap — CHANGES_REQUESTED** [review]
I think this improvement should be moved inside the API or Utils to ensure it's applied also to other components using the WMS capabilities functionalities.

This is the catalog workflow:

- request the records from the WMS capabilities API https://github.com/geosolutions-it/MapStore2/blob/master/web/client/epics/catalog.js#L83
- parse the records from the WMS capabilities https://github.com/geosolutions-it/MapStore2/blob/master/web/client/api/WMS.js#L76-L99
- convert the WMS record to the MapStore layer object https://github.com/geosolutions-it/MapStore2/blob/master/web/client/utils/CatalogUtils.js#L494-L558

@mbarto should we transform in uppercase directly inside the [parsing record](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/api/WMS.js#L80)  step or while [converting to layer object](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/utils/CatalogUtils.js#L524)?

**Comment by MuhweziDeo** [comment]
I personally we should do it while parsing the records

**Comment by MuhweziDeo** [comment]
I personally we should do it while parsing the records

**Comment by allyoucanmap** [comment]
> I personally we should do it while parsing the records

@MuhweziDeo ok, go for parsing record option

File: web/client/api/WMS.js [hunk 1]
```diff
@@ -77,7 +77,7 @@ export const searchAndPaginate = (json = {}, startPosition, maxRecords, text) =>
     const root = (json.WMS_Capabilities || json.WMT_MS_Capabilities || {}).Capability;
     const service = (json.WMS_Capabilities || json.WMT_MS_Capabilities || {}).Service;
     const onlineResource = getOnlineResource(root);
-    const SRSList = root.Layer && (root.Layer.SRS || root.Layer.CRS) || [];
+    const SRSList = root.Layer && (root.Layer.SRS?.map((srs) => srs.toUpperCase()) || root.Layer.CRS?.map((crs)=>crs.toUpperCase())) || [];
```
Comment (allyoucanmap) [inline web/client/api/WMS.js:80]: Please add a unit test for this change.
Note: we could write this .map once, something like this: 
```js
(root.Layer.SRS || root.Layer.CRS)?.map((crs) => crs.toUpperCase())
```

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

File: web/client/api/WMS.js (hunk 1 above)
Comment (MuhweziDeo) [reply web/client/api/WMS.js:80]: Or Nice one @allyoucanmap. Did not know that one

Comment (allyoucanmap) [reply web/client/api/WMS.js:80]: @MuhweziDeo or probably in this case is better:
```js
const SRSList = (root.Layer && (root.Layer.SRS || root.Layer.CRS) || []).map((srs) => srs.toUpperCase());
```

File: web/client/api/__tests__/WMS-test.js [hunk 2]
```diff
@@ -175,6 +174,18 @@ describe('Test correctness of the WMS APIs', () => {
             }
         });
     });
+
+    it('GetRecords transform SRS List to uppercase', (done) => {
+        API.getRecords('base/web/client/test-resources/wms/GetCapabilities-1.3.1.xml', 0, 2, '').then((result) => {
```
Comment (allyoucanmap) [inline web/client/api/__tests__/WMS-test.js:179]: 1.3.0 is the version of the GetCapabilities response. Here we are testing a GetCapabilities 1.3.0 that uses epsg value lowercase so please rename this file `GetCapabilities-1.3.0-epsg-lowercase.xml`.

File: web/client/api/__tests__/WMS-test.js [hunk 3]
```diff
@@ -175,6 +174,18 @@ describe('Test correctness of the WMS APIs', () => {
...
+            try {
+                expect(result.records[0].SRS).toNotEqual(['EPSG:3857', 'EPSG:4326', 'CRS:84']);
```
Comment (allyoucanmap) [inline web/client/api/__tests__/WMS-test.js:181]: Probably you mean to use `toEqual`?

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

File: web/client/api/__tests__/WMS-test.js (hunk 3 above)
Comment (MuhweziDeo) [reply web/client/api/__tests__/WMS-test.js:181]: Sorry forgot this when I was testing to make sure the test fails

**Review by allyoucanmap — APPROVED** [review]

**Comment by tdipisa** [comment]
@ElenaGallo please test this in DEV and let us know if we can do the backport.

**Comment by ElenaGallo** [comment]
@MuhweziDeo @tdipisa there seems to be a a protocol error for the layers present in the service (http://cmein.terragis.net/cgi-bin/mapserv?map=/var/mapdata/countmein/countmein_test.map&VERSION=1.3&REQUEST=getCapabilities&SERVICE=wms)

*How to Reproduce**
- Open [this map]()
- From Options menu, click on Catalog
- Select a layer 
- Add it on map

*Current Result*
The layer is not visible on map and on console this error appears:

> GET https://cmein.terragis.net/cgi-bin/mapserv?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&FORMAT=image%2Fpng&TRANSPARENT=true&LAYERS=ga_president_biden&STYLES=&SRS=EPSG%3A3857&CRS=EPSG%3A3857&TILED=true&map=%2Fvar%2Fmapdata%2Fcountmein%2Fcountmein_test.map&WIDTH=256&HEIGHT=256&BBOX=-16280475.52851626%2C2504688.542848654%2C-15028131.257091934%2C3757032.814272982 net::ERR_SSL_PROTOCOL_ERROR

**Comment by MuhweziDeo** [comment]
> @MuhweziDeo @tdipisa there seems to be a a protocol error for the layers present in the service (http://cmein.terragis.net/cgi-bin/mapserv?map=/var/mapdata/countmein/countmein_test.map&VERSION=1.3&REQUEST=getCapabilities&SERVICE=wms)
> 
> _How to Reproduce_*
> 
> * Open [this map](https://dev.mapstore2.geo-solutions.it/mapstore/#/viewer/openlayers/32940)
> * From Options menu, click on Catalog
> * Select a layer
> * Add it on map
> 
> _Current Result_
> The layer is not visible on map and on console this error appears:
> 
> > GET https://cmein.terragis.net/cgi-bin/mapserv?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&FORMAT=image%2Fpng&TRANSPARENT=true&LAYERS=ga_president_biden&STYLES=&SRS=EPSG%3A3857&CRS=EPSG%3A3857&TILED=true&map=%2Fvar%2Fmapdata%2Fcountmein%2Fcountmein_test.map&WIDTH=256&HEIGHT=256&BBOX=-16280475.52851626%2C2504688.542848654%2C-15028131.257091934%2C3757032.814272982 net::ERR_SSL_PROTOCOL_ERROR

Not sure but I guess this issue more so related to this issue here https://app.zenhub.com/workspaces/mapstore-team-board-5b506fe4624eef429c441528/issues/geosolutions-it/mapstore2/6127 @tdipisa  can confirm on this

**Comment by tdipisa** [comment]
@MuhweziDeo @ElenaGallo yes indeed, it seems a mixed content issue. @ElenaGallo you can test in DEV only using HTTP instead of HTTPS such as for the same map:
http://dev.mapstore2.geo-solutions.it/mapstore/#/viewer/openlayers/32940
becausa HTTPS semme to me not available for cmein.terragis.net/cgi-bin/mapserv

**Comment by ElenaGallo** [comment]
@MuhweziDeo @tdipisa yes, using HTTP for the service `cmein.terragis.net/cgi-bin/mapserv` we have no problems.

**Comment by tdipisa** [comment]
@ElenaGallo then can we ask for the backport?

**Comment by ElenaGallo** [comment]
@MuhweziDeo please backport this to the stable branch

**Comment by tdipisa** [comment]
@MuhweziDeo please provide a backport PR to the stable branch 2021.01.xx. Thank you.

**Pull request merged**